src/openforms/appointments/contrib/base.py

```diff
@@ -0,0 +1,163 @@
+        @param end_at: The end `date` to retrieve available dates for. Default: 14 days after `start_date`.
+        @return: List of `date`
+        """
+        raise NotImplementedError()
+
+    def get_times(
+        self, products: list, location: AppointmentLocation, day: date
+    ) -> list:
+        """
+        Retrieve all available times for given `products`, `location` and `day`.
+
+        @param products: List of `AppointmentProduct`, as obtained from `get_available_products`.
+        @param location: An `AppointmentLocation`, as obtained from `get_locations`.
+        @param day: A `date` to retrieve available times for.
+        @return: List of `datetime`.
+        """
+        raise NotImplementedError()
+
+    def get_calendar(
+        self,
+        products: list,
+        location: AppointmentLocation,
+        start_at: date = None,
+        end_at: date = None,
+    ) -> OrderedDict:
+        """
+        Retrieve a calendar.
+
+        WARNING: This default implementation has significant performance issues.
+        You can override this function with a more efficient implementation if
+        the service supports it.
+
+        @param products: List of `AppointmentProduct`, as obtained from `get_available_products`.
+        @param location: An `AppointmentLocation`, as obtained from `get_locations`.
+        @param start_at: The start `date` to retrieve available dates for. Default: `date.today()`.
+        @param end_at: The end `date` to retrieve available dates for. Default: 14 days after `start_date`.
+        @return: `OrderedDict` where each key represents a date and the values is a list of times.
+        """
+        days = self.get_dates(products, location, start_at, end_at)
+
+        result = OrderedDict()
+
+        try:
+            for day in days:
+                times = self.get_times(products, location, day)
+                result[day] = times
+        except Exception as e:
+            logger.exception(e)
+        finally:
+            return result
+
+    def create_appointment(
+        self, products: list, location: AppointmentLocation, start_at: datetime = None
+    ) -> str:
+        """
+        Create an appointment.
+
+        @param products: List of `AppointmentProduct`, as obtained from `get_available_products`.
+        @param location: An `AppointmentLocation`, as obtained from `get_locations`.
+        @param start_at: A `datetime` to start the appointment, as obtained from `get_calendar`.
+        @return: An appointment identifier as `str`.
+        """
+        raise NotImplementedError()
+
+    def delete_appointment(self, identifier: str) -> bool:
+        """
+        Delete an appointment.
+
+        @param identifier: A `str` that represents the unique identification of the appointment.
+        @return: `True` if succesful, `False` otherwise.
+        """
+        raise NotImplementedError()
```

Comment:
```suggestion
    def delete_appointment(self, identifier: str) -> None:
        """
        Delete an appointment.

        :param identifier: A string that represents the unique identification of the appointment.
        :raises AppointmentDeleteFailed: if the appoinment could not be deleted
        """
        raise NotImplementedError()
```
Returning a boolean is information-poor, instead, plugins can `raise AppointmentDeleteFailed from other_exc` to retain rich information for logging/error-monitoring.